app/main.py

```diff
@@ -2,24 +2,66 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.name = name
+        self.protected_name = f"_{name}"
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"must be in range:"       # noqa
+                             f" {self.min_amount} - {self.max_amount}")
+        setattr(instance, self.protected_name, value)
```

Comment:
It's better to make error messages more descriptive. For example, instead of 'must be in range', you could specify what exactly should be in range.